mediacapture-record/MediaRecorder-canvas-media-source.https.html

```diff
@@ -100,49 +110,61 @@
     return;
   }
 
-  const canvas = document.querySelector("canvas");
-  const ctx = canvas.getContext("2d", {
-    alpha: false,
-  });
-
-  canvas.width = CANVAS_WIDTH;
-  canvas.height = CANVAS_HEIGHT;
-
   const {startCamera, stopCamera} = useUserMedia(CAMERA_CONSTRAINTS);
   startCamera().then(async stream => {
     const videoTrack = stream.getVideoTracks()[0];
-    const { readable: readableStream } = new MediaStreamTrackProcessor({
-      track: videoTrack
-    });
-
-    const composedTrackGenerator = new MediaStreamTrackGenerator({
-      kind: "video"
-    });
-    const sink = composedTrackGenerator.writable;
-
-    ctx.fillStyle = "#333";
-    ctx.fillRect(0, 0, CANVAS_WIDTH, CANVAS_HEIGHT);
-
-    const transformer = new TransformStream({
-      async transform(cameraFrame, controller) {
-        if (cameraFrame && cameraFrame?.codedWidth > 0) {
-          const leftPos = (CANVAS_WIDTH - cameraFrame.displayWidth) / 2;
-          const topPos = (CANVAS_HEIGHT - cameraFrame.displayHeight) / 2;
-
-          ctx.drawImage(cameraFrame, leftPos, topPos);
-
-          const newFrame = new VideoFrame(canvas, {
-            timestamp: cameraFrame.timestamp
-          });
-          cameraFrame.close();
-          controller.enqueue(newFrame);
-        }
+    const worker = await createWorker(`
+      const CANVAS_WIDTH = ` + CANVAS_WIDTH + `;
+      const CANVAS_HEIGHT = ` + CANVAS_HEIGHT + `;
+      onmessage = e => {
+        const videoTrack = e.data;
+        const { readable: readableStream } = new MediaStreamTrackProcessor({
+          track: videoTrack
+        });
+
+        const composedTrackGenerator = new VideoTrackGenerator();
+        const sink = composedTrackGenerator.writable;
```

Comment:
Might it be cleaner to put this worker in a helper like `mediacapture-record/resources/center-crop-worker.js` to follow established patterns like https://github.com/web-platform-tests/wpt/tree/master/fs/resources ?

This would would avoid string code. A follow-up is fine.